ignite/metrics/rouge.py

```diff
@@ -0,0 +1,158 @@
+
+    def _ngramify(self, tokens: List[str], n: int) -> DefaultDict:
+        ngrams: DefaultDict = defaultdict(int)
+        for ngram in (tuple(tokens[i : i + n]) for i in range(len(tokens) - n + 1)):
+            ngrams[ngram] += 1
+        return ngrams
+
+    def _safe_divide(self, numerator: float, denominator: float) -> float:
+        if denominator > 0:
+            return numerator / denominator
+        else:
+            return 0.0
+
+    def _fbeta_score(self, matches: int, recall_total: int, precision_total: int) -> float:
+        recall_score = self._safe_divide(matches, recall_total)
+        precision_score = self._safe_divide(matches, precision_total)
+        denom = (self.beta_sq) * precision_score + recall_score
+        fbeta_score = self._safe_divide((1 + self.beta_sq) * precision_score * recall_score, denom)
+        return fbeta_score
+
+    def _lcs(self, a: List[str], b: List[str]) -> int:
+        if len(a) < len(b):
+            a, b = b, a
+
+        if len(b) == 0:
+            return 0
+
+        row = [0] * len(b)
+        for ai in a:
+            left = 0
+            diag = 0
+            for j, bj in enumerate(b):
+                up = row[j]
+                if ai == bj:
+                    value = diag + 1
+                else:
+                    value = max(left, up)
+                row[j] = value
+                left = value
+                diag = up
+        return left
+
+    def rouge_n(self, y_pred: List[str], y: List[List[str]]) -> float:
+        matches = 0
+        recall_total = 0
+        n = self.n
+        y_pred_dict = self._ngramify(y_pred, n)
+        for model in y:
+            model_dict = self._ngramify(model, n)
+            recall_total += max(len(model) - n + 1, 0)
+            for ngram in y_pred_dict:
+                if model_dict[ngram]:
+                    matches += y_pred_dict[ngram]
+        precision_total = len(y) * max((len(y_pred) - n + 1), 0)
+        fbeta_score = self._fbeta_score(matches, recall_total, precision_total)
+        return fbeta_score
+
+    def rouge_l(self, y_pred: List[str], y: List[List[str]]) -> float:
+        matches = 0
+        recall_total = 0
+        for model in y:
+            matches += int(self._lcs(model, y_pred))
+            recall_total += len(model)
+        precision_total = len(y) * len(y_pred)
+        fbeta_score = self._fbeta_score(matches, recall_total, precision_total)
+        return fbeta_score
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        self._rougetotal = torch.tensor(0, device=self._device)
+        self._num_examples = 0
+        super(Rouge, self).reset()
+
+    @reinit__is_reduced
+    def update(self, output: List[List]) -> None:
+        y_pred, y = output[0], output[1]
+        self._rougetotal = torch.add(self._rougetotal, self.rouge_fn(y_pred, y))
+        self._num_examples += 1
+
+    @sync_all_reduce("_num_examples", "_num_correct")
```

Comment:
In sync_all_reduce we specify class attributes to call all_reduce on. You defined `_num_examples` but there is no `_num_correct`, right ? You can remove it and put `_rougetotal` instead.